storey/steps/flatten.py

```diff
@@ -0,0 +1,34 @@
+from itertools import chain
+
+from storey import Flow, Event
+from storey.dtypes import _termination_obj
+
+
+class Flatten(Flow):
+    """
+    Flattens sequences of sequences (i.e lists of lists, sets of sets, etc...) into a single sequence, by default
+    casts sequence to list. When setting to_set=True, casts sequence to set.
+    """
+
+    def __init__(self, to_set: bool = False, **kwargs):
+        super().__init__(**kwargs)
+        self.to_set: bool = to_set
+
+    async def _do(self, event):
+        if event is _termination_obj:
+            return await self._do_downstream(_termination_obj)
+        else:
+            if self.to_set:
+                flat_elements = set(chain.from_iterable(event.body))
+            else:
+                flat_elements = chain.from_iterable(event.body)
```

Comment:
Why not iterate over `event.body`? looks like from_iterable will do `['ABC', 'DEF'] --> A B C D E F` doesn't look like something we want

Reply:
hmm flattening `['ABC', 'DEF']` will definitely result in that, but why would you do that? You should use Flatten for `[['ABC'], ['ABC']] --> 'ABC' 'DEF'` or `[[1,2,3], [4,5,6], [7,8,9]] --> 1 2 3 4 5 6 7 8 9`.
Flatten is usually used for flattening a sequence of sequences...

Reply:
I mean that's by design... this is how flatten usually works